plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -0,0 +1,394 @@
+    {
+        try {
+            sessionCatalog.renameTable(convert(session), toIdentifier(from), toIdentifier(to));
+        }
+        catch (RESTException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, format("Failed to rename table %s to %s", from, to), e);
+        }
+    }
+
+    @Override
+    public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        try {
+            return tableCache.computeIfAbsent(
+                    schemaTableName.toString(),
+                    key -> {
+                        BaseTable baseTable = (BaseTable) sessionCatalog.loadTable(convert(session), toIdentifier(schemaTableName));
+                        return new BaseTable(baseTable.operations(), quotedTableName(schemaTableName));
+                    });
+        }
+        catch (NoSuchTableException e) {
+            throw new TableNotFoundException(schemaTableName, e);
+        }
+        catch (Exception e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, format("Failed to load table: %s", schemaTableName), e);
+        }
+    }
+
+    @Override
+    public void updateTableComment(ConnectorSession session, SchemaTableName schemaTableName, Optional<String> comment)
+    {
+        Table icebergTable = sessionCatalog.loadTable(convert(session), toIdentifier(schemaTableName));
+        if (comment.isEmpty()) {
+            icebergTable.updateProperties().remove(TABLE_COMMENT).commit();
+        }
+        else {
+            icebergTable.updateProperties().set(TABLE_COMMENT, comment.get()).commit();
+        }
+    }
+
+    @Override
+    public String defaultTableLocation(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        String tableName = createNewTableName(schemaTableName.getTableName());
+
+        Map<String, String> properties = sessionCatalog.loadNamespaceMetadata(convert(session), Namespace.of(schemaTableName.getSchemaName()));
+        String databaseLocation = properties.get(IcebergSchemaProperties.LOCATION_PROPERTY);
+        checkArgument(databaseLocation != null, "location must be set for %s", schemaTableName.getSchemaName());
+
+        return join("/", databaseLocation, tableName);
+    }
+
+    private String createNewTableName(String baseTableName)
+    {
+        String tableName = baseTableName;
+        if (useUniqueTableLocation) {
+            tableName += "-" + randomUUID().toString().replace("-", "");
+        }
+        return tableName;
+    }
+
+    @Override
+    public void setTablePrincipal(ConnectorSession session, SchemaTableName schemaTableName, TrinoPrincipal principal)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "setTablePrincipal is not supported for Iceberg REST catalog");
+    }
+
+    @Override
+    public void createView(ConnectorSession session, SchemaTableName schemaViewName, ConnectorViewDefinition definition, boolean replace)
+    {
```

Comment:
This has been a bit of an issue since all the tests expect a very specific message.  I'll update this because it brings it inline with the other unsupported operations, but will probably need to override the related tests as well.